TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -45,9 +46,76 @@ private SubSelect CreateSubSelect(CodeElementsParser.SubselectContext context)
             {
                 selectClause = CreateSelectClause(context.sql_selectClause());
             }
-
+            if (context.from_clause() != null)
+            {
+                fromClause = CreateFromClause(context.from_clause());
+            }
             return new SubSelect(selectClause, fromClause);
+        }
+
+        private FromClause CreateFromClause(CodeElementsParser.From_clauseContext context)
+        {
+            var tableReferences = new List<SingleTableReference>();
+            if (context.table_references() != null)
+            {
+                foreach (var tableContext in context.table_references().table_reference())
+                {
+                    SingleTableReference tableReference = CreateSingleTableOrViewReferences(tableContext);
+                    tableReferences.Add(tableReference);
+                }
+
+                return new FromClause(tableReferences);
+            }
+
+            return null;
+        }
+
+        public SingleTableReference CreateSingleTableOrViewReferences(CodeElementsParser.Table_referenceContext context)
```

Comment:
`private` method + rename to `CreateSingleTableOrViewReference`, as the method creates a single instance.

Reply:
Note that this method will soon evolve into two distinct methods because a `Table_referenceContext` can lead to the creation of single-table-or-view-reference and many other forms of references.